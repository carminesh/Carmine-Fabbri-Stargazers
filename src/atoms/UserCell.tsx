import { StyleSheet, Text, View, useWindowDimensions, ScaledSize, Image } from 'react-native';
import React, { memo } from 'react';
import ColorsPalette from '../constants/ColorsPalette';

interface UserCellProps {
    login: string;
    id: number;
    avatar_url: string;
}

const UserCell: React.FC<UserCellProps> = ({ login, id, avatar_url }) => {
    const dimensions = useWindowDimensions();
    const styles = style(dimensions);

    return (
        <View style={styles.cellContainer}>
            <Image source={{ uri: avatar_url }} style={styles.avatar} />
            <View style={styles.infosContainer}>
                <Text style={styles.textHeader}>{login}</Text>
                <Text>{login}</Text>
            </View>
        </View>
    );
};

export default memo(UserCell);

const style = (dimensions: ScaledSize) =>
    StyleSheet.create({
        cellContainer: {
            alignSelf: 'center',
            width: dimensions.width - 32,
            height: 76,
            borderRadius: 5,
            backgroundColor: ColorsPalette.WHITE,
            marginBottom: 10,
            paddingHorizontal: 18,
            paddingVertical: 10,
            flexDirection: 'row',
            alignItems: 'center',
        },
        infosContainer: {
            alignItems: 'center',
            paddingVertical: 10,
            paddingHorizontal: 14,
        },
        avatar: {
            width: 40,
            height: 40,
            borderRadius: 40,
        },
        textHeader: {
            fontSize: 16,
            color: ColorsPalette.BLUE_ACCENT,
            fontWeight: 'normal',
        },
    });
