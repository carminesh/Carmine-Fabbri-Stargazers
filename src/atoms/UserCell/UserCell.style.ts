import ColorsPalette from "../../constants/ColorsPalette";
import { ScaledSize, StyleSheet } from 'react-native';

export const style = (dimensions: ScaledSize) =>
    StyleSheet.create({
        cellContainer: {
            alignSelf: 'center',
            width: dimensions.width - 32,
            height: 76,
            borderRadius: 5,
            backgroundColor: ColorsPalette.WHITE,
            marginBottom: 10,
            paddingHorizontal: 18,
            paddingVertical: 10,
            flexDirection: 'row',
            alignItems: 'center',
        },
        infosContainer: {
            alignItems: 'center',
            paddingVertical: 10,
            paddingHorizontal: 14,
        },
        avatar: {
            width: 44,
            height: 44,
            borderRadius: 40,
        },
        textHeader: {
            fontSize: 16,
            color: ColorsPalette.BLUE_ACCENT,
            fontWeight: 'normal',
        },
    });
