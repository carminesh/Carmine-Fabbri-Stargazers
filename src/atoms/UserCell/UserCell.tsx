import { Text, View, useWindowDimensions, Image } from 'react-native';
import React, { memo } from 'react';
import { style } from './UserCell.style';

interface UserCellProps {
    login: string;
    id: number;
    avatar_url: string;
}

const UserCell: React.FC<UserCellProps> = ({ login, id, avatar_url }) => {
    const dimensions = useWindowDimensions();
    const styles = style(dimensions);

    return (
        <View style={styles.cellContainer}>
            <Image testID="user-cell-avatar" source={{ uri: avatar_url }} style={styles.avatar} resizeMode={'contain'} />
            <View style={styles.infosContainer}>
                <Text>
                    Username: <Text style={styles.textHeader}>{login}</Text>
                </Text>
            </View>
        </View>
    );
};

export default memo(UserCell);
