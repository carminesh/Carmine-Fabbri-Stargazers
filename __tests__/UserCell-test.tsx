import 'react-native';
import React from 'react';
import { render } from '@testing-library/react-native';
import UserCell from '../src/atoms/UserCell/UserCell';
import renderer from 'react-test-renderer';

describe('UserCell testing', () => {
    it('renders correctly', () => {
        const tree = renderer.create(<UserCell login="johndoe" id={1} avatar_url="https://example.com/avatar.png" />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should display the username passed as a prop', async () => {
        const { getByText } = render(<UserCell login="johndoe" id={1} avatar_url="https://example.com/avatar.png" />);

        const username = getByText('Username: johndoe');
        expect(username).toBeTruthy();
    });

    it('should display the avatar passed as a prop', async () => {
        const { getByTestId } = render(<UserCell login="johndoe" id={1} avatar_url="https://example.com/avatar.png" />);

        const avatar = getByTestId('user-cell-avatar');
        expect(avatar).toBeTruthy();
    });
});
