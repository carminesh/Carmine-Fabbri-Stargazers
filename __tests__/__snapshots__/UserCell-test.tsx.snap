// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserCell testing renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "alignSelf": "center",
      "backgroundColor": "#FFFFFF",
      "borderRadius": 5,
      "flexDirection": "row",
      "height": 76,
      "marginBottom": 10,
      "paddingHorizontal": 18,
      "paddingVertical": 10,
      "width": 718,
    }
  }
>
  <Image
    resizeMode="contain"
    source={
      {
        "uri": "https://example.com/avatar.png",
      }
    }
    style={
      {
        "borderRadius": 40,
        "height": 44,
        "width": 44,
      }
    }
    testID="user-cell-avatar"
  />
  <View
    style={
      {
        "alignItems": "center",
        "paddingHorizontal": 14,
        "paddingVertical": 10,
      }
    }
  >
    <Text>
      Username: 
      <Text
        style={
          {
            "color": "#2C68F7",
            "fontSize": 16,
            "fontWeight": "normal",
          }
        }
      >
        johndoe
      </Text>
    </Text>
  </View>
</View>
`;
